src/routes/booking.routes.js

```diff
@@ -0,0 +1,74 @@
+import { Router } from "express";
+import { body, param, query } from "express-validator";
+import { validateJwt } from "../middleware/validate-jwt.js";
+import { validateRequestParams } from "../middleware/validate-request-params.js";
+import {
+  bookingsGet,
+  getBookingById,
+  putBooking,
+  bookingDelete,
+  bookingPost,
+} from "../controller/booking.controller.js";
+
+const router = Router();
+
+router.get(
+  "/",
+  [
+    query("limit")
+      .optional()
+      .isNumeric()
+      .withMessage("Limit must be a numeric value"),
+    query("page") 
+      .optional()
+      .isNumeric()
+      .withMessage("Page must be a numeric value"),
+    validateRequestParams,
+  ],
+  bookingsGet,
+);
+
+router.get(
+  "/:id",
+  [param("id").isMongoId(), validateRequestParams],
+  getBookingById,
+);
+
+router.put(
+  "/:id",
+  [
+    validateJwt,
+    param("id").isMongoId(),
+    body("date_start", "Start date is required").notEmpty().optional(), // Cambiado a opcional
+    body("date_end", "End date is required").notEmpty().optional(), // Cambiado a opcional
+    body("tp_status").isEmpty().withMessage("tp_status is not allowed"), // Validar que tp_status no sea proporcionado por el usuario
+    validateRequestParams,
+  ],
+  putBooking,
+);
```

Comment:
esto esta mal implemetado, primero, optional deberia ser el primer metodo, segundo, de ser definido, tienes que validar que sea una fecha valida por ejemplo.
```suggestion
router.put(
  "/:id",
  [
    validateJwt,
    param("id").isMongoId(),
    body("date_start", "If defined, start date must be defined as YYYY-MM-DD").optional().custom((date) => {
      if (!date typeof string) throw new Error("Date must be a string");
      if (date.split("-").length === 3) throw new Error("Date not valid");
      const [year, month, day] = date.split("-");
      const isValidDate = Date,validate(year, month, day)
      
      if (!isValidDate) throw new Error("Date not valid");
    })
    // ... demas validaciones
    body("tp_status").isEmpty().withMessage("tp_status is not allowed"), // Validar que tp_status no sea proporcionado por el usuario
    validateRequestParams,
  ],
  putBooking,
);
```

ese es solo un ejemplo de como hacerlo, tambien te podrias apoyar con otros metodos ya definidos, lo que quiero es que dejes de solo validar `notEmpty`. puedes guiarte de la [documentacion](https://express-validator.github.io/docs/guides/validation-chain) y tambien te pudes guiar de este proyecto, y [este](https://github.com/bcastillo-2022474/gestion-ventas/blob/master/src/category/category.routes.js) ejemplo en particular y [este](https://github.com/bcastillo-2022474/gestion-ventas/blob/master/src/products/product.routes.js), tambien echale un ojo a [este](https://github.com/bcastillo-2022474/gestion-ventas/blob/master/src/user/user.routes.js), lo que si me esta gustando bastante es el metodo de `withMessage(message)`, sigue utilizando ese metodo para definir los mensajes de nuestros middlewares. tambien te recomiendo que mires en la [doc oficial](https://express-validator.github.io/docs/api/validation-chain/#built-in-validators) en donde estan todos los metodos que podes usar, y toma en cuenta que el orden en el que llamas a los validadores importa, [leer esto](https://express-validator.github.io/docs/guides/validation-chain#chaining-order), es super important